addons/controllers/clusterbootstrap_controller.go

```diff
@@ -677,20 +1101,33 @@ func (r *ClusterBootstrapReconciler) getCNI(
 		return nil, err
 	}
 	if selectedCNI != "" {
+		var getPkgMetadataErrs []error
+		foundCNI := false
 		for _, cni := range clusterBootstrapTemplate.Spec.CNIs {
-			if selectedCNI == packageShortName(cni.RefName) {
+			// Package should be available in cluster namespace
+			pkgRefName, _, getPkgMetadataErr := util.GetPackageMetadata(r.context, r.Client, cni.RefName, cluster.Namespace)
```

Comment:
Wouldn't using just ```err``` be easier to read than ```getPkgMetadataErr```?

Reply:
there are few reasons: 
1. Shadowing: using `err` everywhere is simple, but it could shadow previous assigned `err`.
2. Self-Explanation: Using a self-explanation variable could help reader understand what kind of error it would hold, and simplifies the error type check. I.e., `if errors.IsNotFound(err)`
3. Scope: This scope of above error in particular is only for L1103-L1123, that error is not supposed to be a global accessible error.

Reply:
If defined correctly it wouldn't shadow previous definitions. In my opinion using a single variable err is more readable:) But that is more programmer's choice I guess